### PR TITLE
[Delivered #187355059]Users should be able to view and edit their pro…

### PR DESCRIPTION
**Pull request by pacifiquemboni**
## 1.What does this PR do?
Users should be able to view and edit their profiles + billing address
# ⚠️⚠️How to test this PR
1. Clone `develop` branch and checkout to `user-profile-settings2`
2. Install all required dependencies `npm install`
3. Follow `.env.example` to create your `.env` file
4. Start the application using `npm start`
5.  Navigate to login and start by signing up, verify your account 
  `N.B:`  during verification you will not see grafical UI but after clicking button to verify check if you recieve another email for successfully verification and go ba back to your application try to login
6. After login navigate to `profile` link in menu and update your personal information and billing address

## 2.Description
I want to get my profile created once I register for the application.
I want my profile to display my:
 ### 1.Personal Information
   -name, email,gender, birthdate,
   -preferred language, preferred currency,where you live,
 ### 2.Billing Address (check the right fields of the billing address)

Screenshot 2024-06-24 114309
Screenshot 2024-06-24 114343

## 3.Acceptance Criteria
I want to be able to Edit my profile, but I should NOT be able to edit another users profile
### Scenario:
Given User registers to be a member of the App.
Registration is done.
User Profile is created.

### CODE REVIEW

File: __tests__/userslice.test.tsx
Comment (UmuhireJessie) [inline __tests__/userslice.test.tsx:1]: Why do you have this file if every code in it is commented?

File: src/components/billingAddress/updatebillingaddress.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import React from "react";
+// import "../styles/style.scss";
+
+// import updateaddressimg from "../../assets/images/address.png";
```
Comment (UmuhireJessie) [inline src/components/billingAddress/updatebillingaddress.tsx]: Try as much as possible to avoid unnecessary codes

**Review by UmuhireJessie — CHANGES_REQUESTED** [review]
@pacifiquemboni :
1. The code looks good but cannot be tested using the deployed link. 
2. Please provide a deployed backend link
Screenshot 2024-06-24 at 18 25 49

**Review by UmuhireJessie — CHANGES_REQUESTED** [review]
Good work @pacifiquemboni, please consider the following:
1. Remove the error after successful update of profile
Screenshot 2024-07-03 at 18 50 46

2. Close the editing modal after sending the request (From the above the modal is still open yet we have the response)

3. I think you can find a better way to show that there is no data. Maybe add some styles on the word and remember to use the same way throughout. 
(As you see from the above, the birthdate is empty yet you haven't used `.....No data` word)

**Review by shema-surge — APPROVED** [review]

**Review by lilika67 — APPROVED** [review]

File: src/components/personalInfo/personalInfo.tsx [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+import nodata from "../../assets/images/nodata.png"
+import { Link } from "react-router-dom";
+import UpdateProfile from "../../views/updateprofile";
+
+const PersonalInfo: React.FC = () => {
+  const dispatch = useDispatch<AppDispatch>();
+  const { user, loading, error, isSuccess } = useSelector(
+    (state: RootState) => state.user
+  );
+  const { id } = useParams<{ id?: string }>();
```
Comment (UmuhireJessie) [inline src/components/personalInfo/personalInfo.tsx:19]: Consider checking why this is still undefined even when the login has succeeded.

**Review by UmuhireJessie — CHANGES_REQUESTED** [review]
@pacifiquemboni please check this out. Why is the user becoming undefined?
Screenshot 2024-07-08 at 17 54 51

**Review by Celestin4 — APPROVED** [review]

**Review by UmuhireJessie — APPROVED** [review]

**Pull request merged**